### PR TITLE
Fix ParamsMap copyWith param aliases

### LINKED ISSUE #3476: The value taken by a parameter is affected by uppercase letters in the parameter name
### Expected behavior and actual behavior

The uppercase and lowercase parameters behave differently in value override. For example, the parameters params.msg and params.msG in the following workflow takes the values from different scopes, just because their names containing uppercase letters or not. 

Expectation: both parameters should behave the same, no matter whether there are uppercase letters in the parameter names.

### Steps to reproduce the problem

file: wf_addparams.nf

```groovy
nextflow.enable.dsl=2

params.msg="workflow msg lower case"
params.msG="workflow msg upper case"

// include the process
include { print_msg } from \
    './msg.nf' addParams (
        msg: "include msg lower case",
        msG: "include msg upper case"
        )


workflow {
    print_msg()
}
```

The included process file is as follows:
file: msg.nf

```groovy
process print_msg {

    echo true

    script:
    """
    echo msg: ${params.msg}
    echo msG: ${params.msG}
    """
}
```

### Program output 

- Run 1

    * Command
       nextflow run ./wf_addparams.nf
    * Output
    N E X T F L O W ~ version 22.04.5 Launching `./wf_addparams.nf` [soggy_swirles] DSL2 - revision: 61b19cc590 WARN: The `echo` directive has been deprecated - use to `debug` instead executor > local (1) [ef/9126b4] process > print_msg [100%] 1 of 1 ✔ msg: include msg lower case msG: workflow msg upper case
    * Problem: as you can see, params.msg takes the value set at addParams, while params.msG takes the global value set at the beginning of the workflow file.

- Run 2

    * Command
       nextflow run ./wf_addparams.nf--msg "cmd msg lower case" --msG "cmd msg upper case"
    * Output
    N E X T F L O W ~ version 22.04.5 Launching `./wf_addparams.nf` [sad_turing] DSL2 - revision: 61b19cc590 WARN: The `echo` directive has been deprecated - use to `debug` instead executor > local (1) [50/bebd58] process > print_msg [100%] 1 of 1 ✔ msg: include msg lower case msG: cmd msg upper case
    * Problem: again, params.msg takes the value set at addParams, while params.msG takes the value set at the command line.

### Environment 

* Nextflow version: 22.04.5
* Java version: openjdk version "17.0.5"
* Operating system: 18.04.1-Ubuntu
* Bash version: GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)

### PR DESCRIPTION
**Pull request by bentsherman**
Close #3476 

When updating a params map, the aliases weren't being removed from the read-only set, which prevented the params map from being updated correctly.

### CODE REVIEW

**Comment by pditommaso** [comment]
it should include some unit tests to validate the solution

**Comment by bentsherman** [comment]
One more bug fix for the stable release? 😄

**Comment by marcodelapierre** [comment]
@pditommaso able to review over the next week? shall I help?

**Review by marcodelapierre — APPROVED** [review]
Local build and test successful.

Also tested the examples provided in the bug report in the root issue https://github.com/nextflow-io/nextflow/issues/3476 .

I only have one question Ben. In this case:
```
nextflow run ./wf_addparams.nf --msg "cmd msg lower case" --msG "cmd msg upper case"
```

It is correct that the winning set of parameters is the one coming from the `addParams` at the `include`, see below, NOT the cmd line ones, right?

```
params.msg="workflow msg lower case"
params.msG="workflow msg upper case"

// include the process
include { print_msg } from \
    './msg.nf' addParams (
        msg: "include msg lower case",
        msG: "include msg upper case"
        )
```

If this is OK, then good to merge for me.

**Comment by bentsherman** [comment]
> It is correct that the winning set of parameters is the one coming from the addParams at the include, see below, NOT the cmd line ones, right?

Yes, the addParams should take priority in this case

**Comment by marcodelapierre** [comment]
@pditommaso  ready to merge IMO

File: modules/nextflow/src/main/groovy/nextflow/script/ScriptBinding.groovy [hunk 1]
```diff
@@ -239,7 +239,11 @@ class ScriptBinding extends WorkflowBinding {
         }
 
         private ParamsMap allowNames(Set names) {
-            readOnlyNames.removeAll(names)
+            for( String name : names ) {
+                def name2 = name.contains('-') ? hyphenToCamelCase(name) : camelCaseToHyphen(name)
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/script/ScriptBinding.groovy:243]:
```suggestion
                final name2 = name.contains('-') ? hyphenToCamelCase(name) : camelCaseToHyphen(name)
```

**Review by pditommaso — APPROVED** [review]

**Pull request merged**